qutip/tests/test_simdiag.py

```diff
@@ -0,0 +1,60 @@
+import pytest
+import numpy as np
+import qutip
+
+
+@pytest.mark.parametrize('num_mat', [1, 2, 3, 5])
+def test_simdiag(num_mat):
+    N = 10
+
+    U = qutip.rand_unitary(N)
+    cummuting_matrices = [U * qutip.qdiags(np.random.rand(N), 0) * U.dag()
+                          for _ in range(num_mat)]
+    all_evals, evecs = qutip.simdiag(cummuting_matrices)
+
+    for matrix, evals in zip(cummuting_matrices, all_evals):
+        for eval, evec in zip(evals, evecs):
+            assert matrix * evec == evec * eval
+
+
+@pytest.mark.parametrize('num_mat', [1, 2, 3, 5])
+def test_simdiag_no_evals(num_mat):
+    N = 10
+
+    U = qutip.rand_unitary(N)
+    cummuting_matrices = [U * qutip.qdiags(np.random.rand(N), 0) * U.dag()
+                          for _ in range(num_mat)]
+    evecs = qutip.simdiag(cummuting_matrices, evals=False)
+
+    for matrix in cummuting_matrices:
+        for evec in evecs:
+            Mvec = matrix * evec
+            eval = Mvec.full()[0,0] / evec.full()[0,0]
```

Comment:
```suggestion
            eval = Mvec.norm() / evec.norm()
```
There is a small chance the first entry is `0` and this is more succinct.